src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore.hidden === false && herbivore instanceof Herbivore) {
+      herbivore.health -= 50;
+      herbivore.checkHealth();
+    }
```

Comment:
[FUNCTION NAMING] - The `hide` method in the Herbivore class correctly sets the `hidden` property to true, but the `bite` method should only check if the `herbivore` is an instance of Herbivore and is not hidden. There's no need to check if `herbivore.hidden` is false, as the `hide` method does not toggle the value but sets it to true. The condition could be simplified to `if (!herbivore.hidden && herbivore instanceof Herbivore) { ... }`.